File: app/src/main/java/com/yhjoo/dochef/model/SearchResult.java
package com.yhjoo.dochef.model;

import com.chad.library.adapter.base.entity.MultiItemEntity;

public class SearchResult<T> implements MultiItemEntity {
    public int itemType;
    public T content;

    public SearchResult(int itemType, T content) {
        this.itemType = itemType;
        this.content = content;
    }

    public SearchResult(int itemType) {
        this.itemType = itemType;
    }

    public T getContent() {
        return content;
    }

    @Override
    public int getItemType() {
        return itemType;
    }
}

File: app/src/main/java/com/yhjoo/dochef/model/SearchResult.kt
package com.yhjoo.dochef.model

import com.chad.library.adapter.base.entity.MultiItemEntity

class SearchResult<T> : MultiItemEntity {
    private var multiType: Int
    var content: T? = null

    constructor(itemType: Int, content: T) {
        this.multiType = itemType
        this.content = content
    }

    constructor(itemType: Int) {
        this.multiType = itemType
    }

    override fun getItemType(): Int {
        return multiType
    }
}